packages/redux/src/entities/schemas/productsList.ts

```diff
@@ -28,6 +28,7 @@ export default new schema.Entity(
       const {
```

Comment:
Here you will need to do the following:

In line 53, replace `facetGroups` with `newFacetGroups` so the `facetId` property of the filterSegments are calculated correctly with the hash. This will help finding the corresponding facet. 

In lines 89 and 102, replace the cast to `FacetGroup` to `FacetGroupWithListingHash` (that is the type that I suggested on the comment for the facet entity schema which you will need to import)

Reply:
Updated, thanks!